tests/integration/relations/test_relations.py

```diff
@@ -0,0 +1,175 @@
+    await ops_test.model.wait_for_idle(
+        status="active",
+        timeout=3000,
+        raise_on_error=False,
+    )
+
+    app = ops_test.model.applications[APP_NAME]
+    async with ops_test.fast_forward():
+        await ops_test.model.relate(APP_NAME, f"{APPLICATION_APP_NAME}:{FIRST_DATABASE_RELATION}")
+        await ops_test.model.block_until(
+            lambda: "blocked" in {unit.workload_status for unit in app.units},
+            timeout=1500,
+        )
+
+    # Sleep for a while to allow the relation to be established.
+    time.sleep(10)
+    await ops_test.model.applications[APP_NAME].destroy_relation(
+        f"{APP_NAME}:{DATABASE_RELATION}", f"{APPLICATION_APP_NAME}:{FIRST_DATABASE_RELATION}"
+    )
+    await ops_test.model.wait_for_idle(
+        status="active",
+        timeout=3000,
+        raise_on_error=False,
+    )
+
+    logger.info(" check database creation 'waltz'")
+    await check_database_creation(ops_test, "waltz")
+
+    finos_waltz_users = [f"relation_id_{relation.id}"]
+    logger.info(f" check database users existence '{finos_waltz_users}'")
+    await check_database_users_existence(ops_test, finos_waltz_users, [])
+
+    logger.info(
+        f" remove relation: {FINOZ_WALTZ_APP_NAME}:{DB_RELATION} - {APP_NAME}:{DB_RELATION}"
+    )
+    await ops_test.model.applications[APP_NAME].remove_relation(
+        f"{FINOZ_WALTZ_APP_NAME}:{DB_RELATION}", f"{APP_NAME}:{DB_RELATION}"
+    )
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+    logger.info(f" check database users not existence '{finos_waltz_users}'")
+    await check_database_users_existence(ops_test, [], finos_waltz_users)
+
+
+@pytest.mark.group(1)
+async def test_modern_endpoint_with_multiple_related_endpoints(ops_test: OpsTest):
+    async with ops_test.fast_forward():
+        await ops_test.model.relate(FINOZ_WALTZ_APP_NAME, f"{APP_NAME}:{DB_RELATION}")
+        await ops_test.model.wait_for_idle(
+            status="active",
+            timeout=3000,
+            raise_on_error=False,
+        )
+
+    app = ops_test.model.applications[APP_NAME]
+    async with ops_test.fast_forward():
+        await ops_test.model.relate(APP_NAME, f"{APPLICATION_APP_NAME}:{FIRST_DATABASE_RELATION}")
+        await ops_test.model.block_until(
+            lambda: "blocked" in {unit.workload_status for unit in app.units},
+            timeout=1500,
+        )
+
+    # Sleep for a while to allow the relation to be established.
+    time.sleep(10)
+    async with ops_test.fast_forward():
+        await ops_test.model.applications[APP_NAME].remove_relation(
+            f"{FINOZ_WALTZ_APP_NAME}:{DB_RELATION}", f"{APP_NAME}:{DB_RELATION}"
+        )
+        await ops_test.model.wait_for_idle(
+            apps=[FINOZ_WALTZ_APP_NAME],
+            status="blocked",
+            raise_on_blocked=False,
+            timeout=1000,
+        )
```

Comment:
`await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)` is missing at this point.

Reply:
@marceloneppel @taurus-forever I simplified the test